### ISSUE TITLE
Link is not generating 

### ISSUE THREAD

**Issue by Devprajapati2003** [LiNuX-Mallu/CAM-DUMPER#17]
From few days .  Termux tools which generate ngrok links have stopped working.  CamPhish,  camdumper, Zphisher , adv phishing etc tools which generate ngrok links are not generating links . 

It is not only my problem.  But this is happening to every Termux user .

Why this is happening.  If you don't believe you can try

**Comment by LiNuX-Mallu** [comment]
iam working on this issue .

**Comment by LiNuX-Mallu** [comment]
Ngrok made a change in structure of public url thats the issue cause .
The link is generating but you can't see it through our tool but you can get
it using another trick .

Iam gonna fix this issue in few days now iam lit busy 
So everyone should use this trick iam writing below until i fix the code .

Commands -
$ apt install jq curl
$ echo "web_addr: 4045" >> $HOME/.ngrok2/ngrok.yml

Now after running the tool you can't probably see direct link
So take a new session and run

$ curl -s http://127.0.0.1:4045/api/tunnels | jq '.tunnels[0].public_url'

So you will get a output which is the direct link .

Thank you .

**Comment by LiNuX-Mallu** [comment]
Issue fixed in the code .

**Comment by Devprajapati2003** [comment]
Ok thanks 
Fix this issue so that we can get directly link using the tool

**Comment by Reaperv2** [comment]
It still doesnt work

**Comment by LiNuX-Mallu** [comment]
> It still doesnt work

Yes that was some production bug and now i did complete fix i guess , you can check and let me know if there is some issue . Remember ' it is necessary to add web_addr into .ngrok2/ngrok.yml

**Comment by Devprajapati2003** [comment]
> > It still doesnt work
> 
> Yes that was some production bug and now i did complete fix i guess , you can check and let me know if there is some issue . Remember ' it is necessary to add web_addr into .ngrok2/ngrok.yml

Please  give all the commands again so that we can check properly

**Comment by Reaperv2** [comment]
Thanks it worked 
![Uploading Screenshot_2021-09-08-14-20-01-539_com.termux.jpg…]()

**Comment by Reaperv2** [comment]
Thanks it worked but in zphisher the link doesnt generate anymore

**Comment by Reaperv2** [comment]
![Uploading Screenshot_2021-09-08-14-20-01-539_com.termux.jpg…]()

**Comment by Reaperv2** [comment]
thanks it worked but it doesnt generate the link in other phishing tools anymore 😂

**Comment by LiNuX-Mallu** [comment]
> > > It still doesnt work
> > 
> > 
> > Yes that was some production bug and now i did complete fix i guess , you can check and let me know if there is some issue . Remember ' it is necessary to add web_addr into .ngrok2/ngrok.yml
> 
> Please give all the commands again so that we can check properly

I ve updated the installation instructions dude please take a look .

**Comment by LiNuX-Mallu** [comment]
> thanks it worked but it doesnt generate the link in other phishing tools anymore

You should use the trick i mentioned above .

**Comment by LiNuX-Mallu** [comment]
> thanks it worked but it doesnt generate the link in other phishing tools anymore

You should use the trick i mentioned above .

**Comment by Reaperv2** [comment]
Thanks it worked 😊

**Comment by Nanji47** [comment]
everything is working fine as on 17-Oct-21 10:28AM. you are a genius man.

**Comment by LiNuX-Mallu** [comment]
> everything is working fine as on 17-Oct-21 10:28AM. you are a genius man.

Tnk you ❤️

**Comment by Anon741** [comment]
The link is showing but facing this issue 
![Screenshot_2021-12-13-19-20-09-65_40deb401b9ffe8e1df2f1cc5ba480b12](https://user-images.githubusercontent.com/84624448/145824689-0e6f47f1-c31f-4604-b763-44076a8ce62d.jpg)

![Screenshot_2021-12-13-19-20-16-79_84d3000e3f4017145260f7618db1d683.jpg](https://user-images.githubusercontent.com/84624448/145825546-65decddc-8cda-4a7a-bd2d-72d2a3408ad6.jpg)

**Comment by LiNuX-Mallu** [comment]
> The link is showing but facing this issue ![Screenshot_2021-12-13-19-20-09-65_40deb401b9ffe8e1df2f1cc5ba480b12](https://user-images.githubusercontent.com/84624448/145824689-0e6f47f1-c31f-4604-b763-44076a8ce62d.jpg)
> 
> ![Screenshot_2021-12-13-19-20-16-79_84d3000e3f4017145260f7618db1d683.jpg](https://user-images.githubusercontent.com/84624448/145825546-65decddc-8cda-4a7a-bd2d-72d2a3408ad6.jpg)

Sign up and then you will get the authtoken there . Then open termux and run $ ./ngrok authtoken <your_token>

**Comment by Harsh793XD** [comment]
It's showing "bash: /data/data/come.termux/files/home/.ngrok2/ngrok.yml: Permission denied"

**Comment by LiNuX-Mallu** [comment]
> It's showing "bash: /data/data/come.termux/files/home/.ngrok2/ngrok.yml: Permission denied"

rm -rf $HOME/.ngrok2
Try this

**Comment by Edul006** [comment]
Dude, please, can you normally explain in order what needs to be written and where, it writes me a (direct link: null) please help.
![Screenshot_2022-05-01-16-37-14-439_com termux](https://user-images.githubusercontent.com/104697392/166146433-981fb855-5f9b-42f2-90fb-abbfce12a645.jpg)
![Screenshot_2022-05-01-16-37-32-726_com termux](https://user-images.githubusercontent.com/104697392/166146435-476d7538-a4f7-4367-931f-59e89060d8fc.jpg)
![Screenshot_2022-05-01-16-37-56-181_com termux](https://user-images.githubusercontent.com/104697392/166146436-3c3926a9-bae5-451d-abe6-3631b802136a.jpg)

**Comment by root-cyborg127** [comment]
Hi bro , Ngrok ippo pazhaya pole work aavunilla can you please add Cloudflare Tunnel Method

**Comment by LiNuX-Mallu** [comment]
> Hi bro , Ngrok ippo pazhaya pole work aavunilla can you please add Cloudflare Tunnel Method

Little busy now, will fix it later

**Comment by Abood99k** [comment]
> everything is working fine as on 17-Oct-21 10:28AM. you are a genius man.

Hi can you help me

**Comment by Rohitkos** [comment]
> From few days . Termux tools which generate ngrok links have stopped working. CamPhish, camdumper, Zphisher , adv phishing etc tools which generate ngrok links are not generating links .
> 
> It is not only my problem. But this is happening to every Termux user .
> 
> Why this is happening. If you don't believe you can try

Brother can you send the all commands step by step for this tool
Because here showing null in the link space

**Comment by Rohitkos** [comment]
> > > > It still doesnt work
> > > 
> > > 
> > > Yes that was some production bug and now i did complete fix i guess , you can check and let me know if there is some issue . Remember ' it is necessary to add web_addr into .ngrok2/ngrok.yml
> > 
> > 
> > Please give all the commands again so that we can check properly
> 
> I ve updated the installation instructions dude please take a look .

Brother can you send the all links step by step because I'm not understanded your sended trick when your sended in starting page step by step commands like that can you send now for the working for this tool plzz plzz brother
